dev/com.ibm.ws.security.authentication/src/com/ibm/ws/security/authentication/cache/CacheObject.java

```diff
@@ -59,7 +69,62 @@ public void addLookupKey(Object key) {
         return lookupKeys;
     }
 
+    /**
+     * Get the {@link Subject} stored in this {@link CacheObject}.
+     *
+     * @return The subject.
+     */
     public Subject getSubject() {
         return this.subject;
     }
+
+    @SuppressWarnings("unchecked")
+    private void readObject(ObjectInputStream input) throws ClassNotFoundException, IOException {
+        /*
+         * Read all non-transient fields.
+         */
+        input.defaultReadObject();
+
+        /*
+         * Read the fields comprising the Subject.
+         */
+        boolean subjectReadOnly = input.readBoolean();
+        Set<Principal> subjectPrincipals = (Set<Principal>) input.readObject();
+        Set<Object> subjectPubCreds = (Set<Object>) input.readObject();
+        Set<Object> subjectPrivCreds = (Set<Object>) input.readObject();
+
+        /*
+         * Create the subject from the subject fields.
+         */
+        subject = new Subject(subjectReadOnly, subjectPrincipals, subjectPubCreds, subjectPrivCreds);
+    }
+
+    private void writeObject(ObjectOutputStream output) throws IOException {
+        /*
+         * Write all non-transient fields.
+         */
+        output.defaultWriteObject();
+
+        /*
+         * Write the fields comprising the Subject.
+         */
+        output.writeBoolean(subject.isReadOnly());
```

Comment:
Should this serialization have a versioning option?

Reply:
I added Serialization in this class for the Beta. I found during testing for this change that I had messed it up. I could version this, but I would only be versioning it to work with a beta release, and instead I intend on future beta releases to recommend clearing the cache before running a new version.

Reply:
I added a serialization version field suitable for future use in the readObject method should serialization change down the road.